python/logview.py

```diff
@@ -105,78 +99,35 @@ def main():
+                logTimestamps=args.timestamped)
+        parser.send(None)
 
         base_time = None
 
-        if not pickle_data_loaded:
-            print "Parsing using the LogFormat: " + `args.timestamped`
-            print "Reading log file..."
-            uavo_parsed = []
-            while fd:
-                try:
-                    if args.timestamped and parser.state == taulabs.uavtalk.UavTalk.STATE_COMPLETE:
-                        # This logging format is somewhat of a hack and simply prepends additional
-                        # information in front of each UAVTalk packet.  We look for this information
-                        # whenever the parser has completed a packet. Note that there is no checksum
-                        # applied to this information so it can be totally messed up, especially if 
-                        # there is a frame shift error. The internal timestamping method of UAVTalk is
-                        # a much better idea.
-
-                        from collections import namedtuple
-                        LogHeader = namedtuple('LogHeader', 'time size')
-
-                        # Read the next log record header
-                        log_hdr_fmt = "<IQ"
-                        log_hdr_data = fd.read(struct.calcsize(log_hdr_fmt))
-
-                        # Check if we hit the end of the file
-                        if len(log_hdr_data) == 0:
-                            # Normal End of File (EOF) at a record boundary
-                            break
-
-                        # Got a log record header.  Unpack it.
-                        log_hdr = LogHeader._make(struct.unpack(log_hdr_fmt, log_hdr_data))
-
-                        # Set the baseline timestamp from the first record in the log file
-                        if base_time is None:
-                            base_time = log_hdr.time
-
-
-                    parser.processByte(ord(fd.read(1)))
-
-                    if parser.state == taulabs.uavtalk.UavTalk.STATE_COMPLETE:
-                        if args.timestamped:
-                            u  = parser.getLastReceivedObject(timestamp=log_hdr.time)
-                        else:
-                            u  = parser.getLastReceivedObject()
-                        if u is not None:
-                            uavo_parsed.append(u)
-
-                except TypeError:
-                    print "End of file"
-                    break
-
-            fd.close()
-
-            print "Processed %d Log File Records" % len(uavo_parsed)
-
-            print "Writing pickled log file to '%s'" % pickle_name
-            import cPickle as pickle
-            pickle_fd = open(pickle_name, 'wb')
-            pickle.dump(githash, pickle_fd)
-            pickle.dump(uavo_parsed, pickle_fd)
-            pickle_fd.close()
-
-        print "Converting log records into python objects"
-        uavo_list = taulabs.uavo_list.UAVOList(uavo_defs)
```

Comment:
Changing this to an ordinary list breaks most of the existing (half-documented) analysis workflow since we rely on the UAVOList.as_numpy_array() method.

Try this out:
```
%run python/calibration/mag_calibration.py
res = mag_calibration(uavo_list.as_numpy_array(UAVO_Magnetometer), uavo_list.as_numpy_array(UAVO_Gyros))
```
Is there a better way to do this now?  If not, just reverting this one line seems to restore previous functionality.